sdk/eventhub/event-hubs/test/internal/createTimerLoop.spec.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as sinon from "sinon";
+import { assert } from "@azure/test-utils";
+import { createTimerLoop } from "../../src/util/timerLoop";
+
+describe("createTimerLoop", function () {
+  let clock: sinon.SinonFakeTimers;
+
+  before(function () {
+    clock = sinon.useFakeTimers();
+  });
+  after(function () {
+    clock.restore();
+  });
+
+  it("loops the exact number of iterations and can be stopped", async function () {
+    const interval = Math.ceil(Math.random() * 1000);
```

Comment:
I don't like randomness in unit tests, since if for some reason the randomness *did* find a failure, it would be hard to debug and understand. If any value in the range is valid, might as well pick one.